tst/EnergyPlus/unit/MixedAir.unit.cc

```diff
@@ -5280,6 +5282,103 @@ TEST_F(EnergyPlusFixture, MixedAir_MiscGetsPart2)
+        "    Zone 2 Outlet Node;      !- Zone Return Air Node or NodeList Name",
+        "Controller:MechanicalVentilation,",
+        "    CM 1,                    !- Name",
+        "    ,                        !- Availability Schedule Name",
+        "    No,                      !- Demand Controlled Ventilation",
+        "    Standard62.1VentilationRateProcedure,  !- System Outdoor Air Method",
+        "    1,                       !- Zone Maximum Outdoor Air Fraction {dimensionless}",
+        "    Zone 1,                  !- Zone or ZoneList 1 Name",
+        "    SZ DSOA,                 !- Design Specification Outdoor Air Object Name 1",
+        "    ,                        !- Design Specification Zone Air Distribution Object Name 1",
+        "    Zone 2,                  !- Zone or ZoneList 2 Name",
+        "    SZ DSOA;                 !- Design Specification Outdoor Air Object Name 2",
+    });
+
+    ASSERT_TRUE(process_idf(idf_objects));
+
+    Real64 SysMassFlow(0.0);        // System supply mass flow rate [kg/s]
+    Real64 OAMassFlow(0.0);         // OA mass flow rate [kg/s]
+    Real64 ExpectedOAMassFlow(0.8); // System design OA flow rate
+    bool ErrorsFound(false);
+
+    GetZoneData(*state, ErrorsFound);
+    GetOARequirements(*state);
+    GetZoneEquipment(*state);
+    GetOAControllerInputs(*state);
+    EXPECT_FALSE(ErrorsFound);
+    EXPECT_EQ(SOAM_VRP, state->dataMixedAir->VentilationMechanical(1).SystemOAMethod);
+
+    state->dataZoneEnergyDemand->ZoneSysEnergyDemand.allocate(2);             // Necessary for CalcMechVentController
+    state->dataSize->SysSizingRunDone = true;                                 // Indicate that a system sizing run has been performed
+    state->dataHeatBal->Zone(1).TotOccupants = 15;                            // Zone 1 total number of people
+    state->dataHeatBal->Zone(2).TotOccupants = 12;                            // Zone 2 total number of people
+    state->dataHeatBal->Zone(1).FloorArea = 1500;                             // Zone 1 total floor area
+    state->dataHeatBal->Zone(2).FloorArea = 500;                              // Zone 2 total floor area
+    state->dataSize->FinalSysSizing.allocate(1);                              // Create instance of system sizing info
+    state->dataSize->FinalSysSizing(1).DesOutAirVolFlow = ExpectedOAMassFlow; // Set design outdoor air flow rate
+    state->dataSize->CurSysNum = 1;                                           // Only one system in this instance
+    state->dataEnvrn->StdRhoAir = 1;                                          // Standard air density assumed to be 1 kg/m3 (simplification)
+    state->dataMixedAir->VentilationMechanical(1).CalcMechVentController(*state, SysMassFlow, OAMassFlow);
+
+    EXPECT_NEAR(
+        ExpectedOAMassFlow, OAMassFlow, 0.001); // Expect to cap the system OA to the desing OA air flow, OAMassFlow without the cap is ~0.86 m3/s
+}
+
```

Comment:
Aim of the test is to check that the system ventilation calculated by the `Controller:MechanicalVentilation` cannot be greater than the system design outdoor air flow.